hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneListStatusHelper.java

```diff
@@ -442,10 +443,19 @@ private void getCacheValues() {
         }
 
         // Copy cache value to local copy and work on it
-        Value copyOmInfo = ObjectUtils.clone(cacheOmInfo);
+        Value copyOmInfo = null;
+        try {
+          if (cacheOmInfo instanceof ICopyObjectInfo) {
+            Method m = cacheOmInfo.getClass().getMethod("copyObject");
+            copyOmInfo = (Value) m.invoke(cacheOmInfo);
+          }
+        } catch (Exception e) {
+          LOG.debug("No such method, use same cache copy");
+        }
```

Comment:
We don't need reflection here.  Simply cast it as below
```java
@@ -442,9 +442,8 @@ private void getCacheValues() {
         }
 
         // Copy cache value to local copy and work on it
-        Value copyOmInfo = ObjectUtils.clone(cacheOmInfo);
-        if (copyOmInfo != null) {
-          cacheOmInfo = copyOmInfo;
+        if (cacheOmInfo instanceof ICopyObject) {
+          cacheOmInfo = ((ICopyObject) cacheOmInfo).copyObject();
         }
         if (StringUtils.isBlank(startKey)) {
           // startKey is null or empty, then the seekKeyInDB="1024/"
```

Reply:
@szetszwo, Thanks for the comments, All comments are fixed.